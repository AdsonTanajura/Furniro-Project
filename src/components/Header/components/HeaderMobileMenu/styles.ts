import styled from 'styled-components';

export const MobileMenuContainer = styled.div.attrs<{ $headerHeight?: number }>(
  ({ $headerHeight }) => ({
    style: { top: `${$headerHeight}px` },
  })
)`
  position: absolute;
  z-index: 10;
  top: 64px;
  left: 0;
  right: 0;
  background: white;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  padding: 16px;
`;

export const NavContainerMobile = styled.div`
  display: flex;
  justify-content: center;
  @media (min-width: 780px) {
    display: none;
  }
`;
