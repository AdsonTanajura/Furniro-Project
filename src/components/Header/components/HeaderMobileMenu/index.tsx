import React from 'react';
import MobileMenuProps from './Interfaces/MobileMenuProps';
import HeaderNavBar from '../HeaderNavBar';
import { MobileMenuContainer, NavContainerMobile } from './styles';

const HeaderMobileMenu = ({ headerHeight }: MobileMenuProps) => {
  return (
    <MobileMenuContainer $headerHeight={headerHeight}>
      <NavContainerMobile>
        <HeaderNavBar />
      </NavContainerMobile>
    </MobileMenuContainer>
  );
};

export default HeaderMobileMenu;
