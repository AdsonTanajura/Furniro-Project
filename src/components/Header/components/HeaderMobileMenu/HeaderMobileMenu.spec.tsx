import { render } from '@testing-library/react';
import HeaderMobileMenu from '.';
import { MemoryRouter } from 'react-router-dom';
describe('<HeaderMobileMenu />', () => {
  const headerHeight = 150;
  const defaultHeaderHeight = 147;

  it('should render all the elements of the component', () => {
    const { container } = render(
      <MemoryRouter>
        <HeaderMobileMenu headerHeight={headerHeight} />
      </MemoryRouter>
    );

    const mobileMenuContainer = container.firstChild;
    const navContainerMobile = mobileMenuContainer?.firstChild;

    expect(navContainerMobile).toBeInTheDocument();
    expect(navContainerMobile).toBeVisible();
    expect(mobileMenuContainer).toBeInTheDocument();
    expect(mobileMenuContainer).toBeVisible();
  });

  it('should change the top syles based on the headerHeight prop', () => {
    const { container } = render(
      <MemoryRouter>
        <HeaderMobileMenu headerHeight={headerHeight} />
      </MemoryRouter>
    );

    const mobileMenuContainer = container.firstChild;
    expect(mobileMenuContainer).toHaveStyle({ top: `${headerHeight}px` });
  });

  it('should have a top style of 147px if the headerHeight property is not sent', () => {
    const { container } = render(
      <MemoryRouter>
        <HeaderMobileMenu />
      </MemoryRouter>
    );
    const mobileMenuContainer = container.firstChild;

    expect(mobileMenuContainer).toHaveStyle({
      top: `${defaultHeaderHeight}px`,
    });
  });
});
