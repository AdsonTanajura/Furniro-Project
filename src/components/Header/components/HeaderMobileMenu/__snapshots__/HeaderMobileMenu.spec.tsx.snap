// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderMobileMenu /> não deve estar visível quando a largura da janela for 780px ou mais 1`] = `
.c2 {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  width: 100vw;
  max-width: 430px;
}

.c3 {
  text-decoration: none;
  font-size: 16px;
  color: #000000;
  transition: border-bottom 0.2s;
}

.c3:hover {
  border-bottom: 2px #000000 solid;
}

.c0 {
  position: absolute;
  z-index: 10;
  top: 150px;
  left: 0;
  left: 0;
  right: 0;
  background: white;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  padding: 16px;
}

.c1 {
  display: flex;
  justify-content: center;
}

@media (max-width: 780px) {
  .c2 {
    flex-direction: column;
  }
}

@media (min-width: 780px) {
  .c0 {
    display: none;
    visibility: hidden;
  }
}

<div>
  <div
    class="c0"
    data-testid="menu-container-mobile"
  >
    <div
      class="c1"
      data-testid="nav-container-mobile"
    >
      <nav>
        <ul
          class="c2"
        >
          <li>
            <a
              class="c3"
              data-discover="true"
              href="/home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="c3"
              data-discover="true"
              href="/home/shop"
            >
              Shop
            </a>
          </li>
          <li>
            <a
              class="c3"
              data-discover="true"
              href="/a"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="c3"
              data-discover="true"
              href="/home/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
